hermes/bin/src/event/queue.rs

```diff
@@ -5,16 +5,21 @@ use std::{
         mpsc::{Receiver, Sender},
         Arc,
     },
-    thread,
+    thread::{self, JoinHandle},
 };
 
+use once_cell::sync::OnceCell;
+
 use super::{HermesEvent, HermesEventPayload, TargetApp, TargetModule};
 use crate::{
     app::{HermesAppName, IndexedApps},
     runtime_context::HermesRuntimeContext,
     wasm::module::{Module, ModuleId},
 };
 
+/// Singleton instance of the Hermes event queue.
+static EVENT_QUEUE_INTANCE: OnceCell<HermesEventQueue> = OnceCell::new();
```

Comment:
Suggestion: if we use `Lazy` (since `LazyLock` is still unstable) we can make the `get_instance` method to return the queue instance directly instead of a `Result`. That way every runtime extension that wishes to use the event queue won't need to check for not initialized errors.

Reply:
But the key problem here, that you cannot initialise it in a random place, for initialisation you need to provide a list of apps, which can not be done by the runtime extensions

Reply:
Hum... got it.